### PR TITLE
fix arg order in merge cmd

### PR DESCRIPTION
**Pull request by johnnyaug**

### CODE REVIEW

**Comment by ozkatz** [comment]
There are a few more breaking changes that I'd like to make to the cli. We should probably batch them together.

**Review by itaiad200 — APPROVED** [review]
Approved, but sync with @ozkatz before you pull for other breaking changes to the CLI.

File: catalog/rocks/cataloger.go [hunk 1]
```diff
@@ -592,12 +592,12 @@ func listDiffHelper(it EntryDiffIterator, limit int, after string) (catalog.Diff
 	return diffs, hasMore, nil
 }
 
-func (c *cataloger) Merge(ctx context.Context, repository string, leftBranch string, rightBranch string, committer string, message string, metadata catalog.Metadata) (*catalog.MergeResult, error) {
+func (c *cataloger) Merge(ctx context.Context, repository string, theirs string, ours string, committer string, message string, metadata catalog.Metadata) (*catalog.MergeResult, error) {
```
Comment (itaiad200) [inline catalog/rocks/cataloger.go:602]: use branch in var name.

File: catalog/rocks/entry_catalog.go [hunk 2]
```diff
@@ -354,20 +354,20 @@ func (e *EntryCatalog) Revert(ctx context.Context, repositoryID graveler.Reposit
 	return e.Store.Revert(ctx, repositoryID, branchID, ref, parentNumber, commitParams)
 }
 
-func (e *EntryCatalog) Merge(ctx context.Context, repositoryID graveler.RepositoryID, from graveler.Ref, to graveler.BranchID, commitParams graveler.CommitParams) (graveler.CommitID, graveler.DiffSummary, error) {
+func (e *EntryCatalog) Merge(ctx context.Context, repositoryID graveler.RepositoryID, theirs graveler.BranchID, ours graveler.Ref, commitParams graveler.CommitParams) (graveler.CommitID, graveler.DiffSummary, error) {
 	if commitParams.Message == "" {
-		commitParams.Message = fmt.Sprintf("Merge '%s' into '%s'", from, to)
+		commitParams.Message = fmt.Sprintf("Merge '%s' into '%s'", ours, theirs)
```
Comment (itaiad200) [inline catalog/rocks/entry_catalog.go:359]: Isn't this the other way around?

File: catalog/rocks/entry_catalog.go [hunk 3]
```diff
@@ -354,20 +354,20 @@ func (e *EntryCatalog) Revert(ctx context.Context, repositoryID graveler.Reposit
 	return e.Store.Revert(ctx, repositoryID, branchID, ref, parentNumber, commitParams)
 }
 
-func (e *EntryCatalog) Merge(ctx context.Context, repositoryID graveler.RepositoryID, from graveler.Ref, to graveler.BranchID, commitParams graveler.CommitParams) (graveler.CommitID, graveler.DiffSummary, error) {
+func (e *EntryCatalog) Merge(ctx context.Context, repositoryID graveler.RepositoryID, theirs graveler.BranchID, ours graveler.Ref, commitParams graveler.CommitParams) (graveler.CommitID, graveler.DiffSummary, error) {
```
Comment (itaiad200) [inline catalog/rocks/entry_catalog.go:357]: How did the types change?

File: cmd/lakectl/cmd/merge.go [hunk 4]
```diff
@@ -41,14 +41,14 @@ var mergeCmd = &cobra.Command{
 	),
 	Run: func(cmd *cobra.Command, args []string) {
 		client := getClient()
-		rightRefURI := uri.Must(uri.Parse(args[0]))
-		leftRefURI := uri.Must(uri.Parse(args[1]))
+		sourceRef := uri.Must(uri.Parse(args[0]))
```
Comment (itaiad200) [inline cmd/lakectl/cmd/merge.go:44]: Why another terminology? Let's stick with either source/destination or theirs/ours (I like the first one better but that's just me)

File: graveler/graveler.go [hunk 5]
```diff
@@ -1138,13 +1138,13 @@ func (g *Graveler) Revert(ctx context.Context, repositoryID RepositoryID, branch
 	return c.ID, c.Summary, nil
 }
 
-func (g *Graveler) Merge(ctx context.Context, repositoryID RepositoryID, from Ref, to BranchID, commitParams CommitParams) (CommitID, DiffSummary, error) {
-	res, err := g.branchLocker.MetadataUpdater(ctx, repositoryID, to, func() (interface{}, error) {
+func (g *Graveler) Merge(ctx context.Context, repositoryID RepositoryID, theirs BranchID, ours Ref, commitParams CommitParams) (CommitID, DiffSummary, error) {
+	res, err := g.branchLocker.MetadataUpdater(ctx, repositoryID, theirs, func() (interface{}, error) {
```
Comment (itaiad200) [inline graveler/graveler.go:1142]: Did you switch the lock on purpose?

**Review by itaiad200 — CHANGES_REQUESTED** [review]
That is one confusing referencing shenanigans! 
I think you got some ours/theirs references mixed-up.

File: catalog/rocks/fake_graveler_test.go [hunk 6]
```diff
@@ -163,7 +163,7 @@ func (g *FakeGraveler) Revert(_ context.Context, _ graveler.RepositoryID, _ grav
 	panic("implement me")
 }
 
-func (g *FakeGraveler) Merge(ctx context.Context, repositoryID graveler.RepositoryID, from graveler.Ref, to graveler.BranchID, _ graveler.CommitParams) (graveler.CommitID, graveler.DiffSummary, error) {
+func (g *FakeGraveler) Merge(ctx context.Context, repositoryID graveler.RepositoryID, theirs graveler.BranchID, ours graveler.Ref, _ graveler.CommitParams) (graveler.CommitID, graveler.DiffSummary, error) {
```
Comment (itaiad200) [inline catalog/rocks/fake_graveler_test.go:166]: Forgot one.

File: graveler/graveler.go [hunk 7]
```diff
@@ -442,14 +441,14 @@ type CommittedManager interface {
 	// This is similar to a two-dot diff in git (left..right)
 	Diff(ctx context.Context, ns StorageNamespace, left, right MetaRangeID) (DiffIterator, error)
 
-	// Compare returns the difference between 'theirs' and 'ours', relative to a merge base 'base'.
+	// Compare returns the difference between 'source' and 'destinationn', relative to a merge base 'base'.
```
Comment (itaiad200) [inline graveler/graveler.go:444]:
```suggestion
	// Compare returns the difference between 'source' and 'destination', relative to a merge base 'base'.
```

**Review by itaiad200 — APPROVED** [review]
LGTM!

**Pull request merged**